example/weather/services/tester/run_tests.go

```diff
@@ -0,0 +1,155 @@
+package tester
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"runtime/debug"
+	"strings"
+	"sync"
+	"time"
+
+	gentester "goa.design/clue/example/weather/services/tester/gen/tester"
+	"goa.design/clue/log"
+	"golang.org/x/exp/slices"
+)
+
+// Ends a test by calculating duration and appending teh results to the test collection
+func endTest(tr *gentester.TestResult, start time.Time, tc *TestCollection, results []*gentester.TestResult) {
+	elapsed := time.Since(start).Milliseconds()
+	tr.Duration = elapsed
+	results = append(results, tr)
+	tc.AppendTestResult(results...)
+}
+
+// recovers from a panicked test. This is used to ensure that the test
+// suite does not crash if a test panics.
+func recoverFromTestPanic(ctx context.Context, testName string, testCollection *TestCollection) {
+	if r := recover(); r != nil {
+		msg := fmt.Sprintf("[Panic Test]: %v", testName)
+		err := errors.New(msg)
+		log.Errorf(ctx, err, fmt.Sprintf("%v", r))
+		// This doesn't work as I'd like because it only prints to stderr, not to a string
+		// 	that I can caputre and put in the goa log.
+		debug.PrintStack()
+		resultMsg := fmt.Sprintf("%v | %v", msg, r)
+		testCollection.AppendTestResult(&gentester.TestResult{
+			Name:     testName,
+			Passed:   false,
+			Error:    &resultMsg,
+			Duration: -1,
+		})
+	}
+}
+
+// Runs the tests from the testmap and handles filtering/exclusion of tests
+func (svc *Service) runTests(ctx context.Context, p *gentester.TesterPayload, testCollection *TestCollection, testMap map[string]func(context.Context, *TestCollection)) (res *gentester.TestResults, err error) {
+	retval := gentester.TestResults{}
```

Comment:
The returned value is already name so the code should either reuse the same name of not name the return values (probably the better option here). i.e. replace `(res *gentester.TestResults, err error)` above with `(*gentester.TestResults, error)`.